niff-1.02/niffsdk1.02/niffio/doc/man/clt.man
.rn '' }`
''' $RCSfile$$Revision$$Date$
'''
''' $Log$
'''
.de Sh
.br
.if t .Sp
.ne 5
.PP
\fB\\$1\fR
.PP
..
.de Sp
.if t .sp .5v
.if n .sp
..
.de Ip
.br
.ie \\n(.$>=3 .ne \\$3
.el .ne 3
.IP "\\$1" \\$2
..
.de Vb
.ft CW
.nf
.ne \\$1
..
.de Ve
.ft R

.fi
..
'''
'''
'''     Set up \*(-- to give an unbreakable dash;
'''     string Tr holds user defined translation string.
'''     Bell System Logo is used as a dummy character.
'''
.tr \(*W-|\(bv\*(Tr
.ie n \{\
.ds -- \(*W-
.ds PI pi
.if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\" diablo 12 pitch
.ds L" ""
.ds R" ""
.ds L' '
.ds R' '
'br\}
.el\{\
.ds -- \(em\|
.tr \*(Tr
.ds L" ``
.ds R" ''
.ds L' `
.ds R' '
.ds PI \(*p
'br\}
.\"	If the F register is turned on, we'll generate
.\"	index entries out stderr for the following things:
.\"		TH	Title 
.\"		SH	Header
.\"		Sh	Subsection 
.\"		Ip	Item
.\"		X<>	Xref  (embedded
.\"	Of course, you have to process the output yourself
.\"	in some meaninful fashion.
.if \nF \{
.de IX
.tm Index:\\$1\t\\n%\t"\\$2"
..
.nr % 0
.rr F
.\}
.TH clt 3 "niffio" "14/Jun/96" "NIFFIO Documentation"
.IX Title "clt 3"
.UC
.if n .hy 0
.if n .na
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.de CQ          \" put $1 in typewriter font
.ft CW
'if n "\c
'if t \\&\\$1\c
'if n \\&\\$1\c
'if n \&"
\\&\\$2 \\$3 \\$4 \\$5 \\$6 \\$7
'.ft R
..
.\" @(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2
.	\" AM - accent mark definitions
.bd B 3
.	\" fudge factors for nroff and troff
.if n \{\
.	ds #H 0
.	ds #V .8m
.	ds #F .3m
.	ds #[ \f1
.	ds #] \fP
.\}
.if t \{\
.	ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.	ds #V .6m
.	ds #F 0
.	ds #[ \&
.	ds #] \&
.\}
.	\" simple accents for nroff and troff
.if n \{\
.	ds ' \&
.	ds ` \&
.	ds ^ \&
.	ds , \&
.	ds ~ ~
.	ds ? ?
.	ds ! !
.	ds /
.	ds q
.\}
.if t \{\
.	ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.	ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.	ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.	ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.	ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.	ds ? \s-2c\h'-\w'c'u*7/10'\u\h'\*(#H'\zi\d\s+2\h'\w'c'u*8/10'
.	ds ! \s-2\(or\s+2\h'-\w'\(or'u'\v'-.8m'.\v'.8m'
.	ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.	ds q o\h'-\w'o'u*8/10'\s-4\v'.4m'\z\(*i\v'-.4m'\s+4\h'\w'o'u*8/10'
.\}
.	\" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds v \\k:\h'-(\\n(.wu*9/10-\*(#H)'\v'-\*(#V'\*(#[\s-4v\s0\v'\*(#V'\h'|\\n:u'\*(#]
.ds _ \\k:\h'-(\\n(.wu*9/10-\*(#H+(\*(#F*2/3))'\v'-.4m'\z\(hy\v'.4m'\h'|\\n:u'
.ds . \\k:\h'-(\\n(.wu*8/10)'\v'\*(#V*4/10'\z.\v'-\*(#V*4/10'\h'|\\n:u'
.ds 3 \*(#[\v'.2m'\s-2\&3\s0\v'-.2m'\*(#]
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.ds oe o\h'-(\w'o'u*4/10)'e
.ds Oe O\h'-(\w'O'u*4/10)'E
.	\" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.	\" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.	ds : e
.	ds 8 ss
.	ds v \h'-1'\o'\(aa\(ga'
.	ds _ \h'-1'^
.	ds . \h'-1'.
.	ds 3 3
.	ds o a
.	ds d- d\h'-1'\(ga
.	ds D- D\h'-1'\(hy
.	ds th \o'bp'
.	ds Th \o'LP'
.	ds ae ae
.	ds Ae AE
.	ds oe oe
.	ds Oe OE
.\}
.rm #[ #] #H #V #F C
.SH "NAME"
.IX Header "NAME"
NIFFIO chunk length table routines.
.Ip "\(bu" 4
.IX Item "\(bu"
\fINIFFIOCLTNew()\fR
.Ip "\(bu" 4
.IX Item "\(bu"
\fINIFFIOCLTDelete()\fR
.Ip "\(bu" 4
.IX Item "\(bu"
\fINIFFIOCLTMakeEntry()\fR
.Ip "\(bu" 4
.IX Item "\(bu"
\fINIFFIOCLTMakeDefaultEntries()\fR
.Ip "\(bu" 4
.IX Item "\(bu"
\fINIFFIOCLTLookup()\fR
.Ip "\(bu" 4
.IX Item "\(bu"
\fINIFFIOCLTCount()\fR
.Ip "\(bu" 4
.IX Item "\(bu"
\fINIFFIOCompareFOURCC()\fR
.SH "NIFFIOCLTNew"
.IX Header "NIFFIOCLTNew"
Return a new, empty NIFFIOChunkLengthTable, or null on failure.
.PP
.Vb 2
\&     NIFFIOChunkLengthTable *
\&     NIFFIOCLTNew()
.Ve
.SH "NIFFIOCLTDelete"
.IX Header "NIFFIOCLTDelete"
Free the memory allocated to a chunk length table.
.PP
.Vb 2
\&     void
\&     NIFFIOCLTDelete(NIFFIOChunkLengthTable *pclt)
.Ve
.SH "NIFFIOCLTMakeEntry"
.IX Header "NIFFIOCLTMakeEntry"
Make an entry into a chunk length table.
.PP
.Vb 3
\&     RIFFIOSuccess
\&     NIFFIOCLTMakeEntry(NIFFIOChunkLengthTable *pclt, 
\&                        niffChklentabEntry cltEntry)
.Ve
.Sh "\s-1ENTRY\s0"
.IX Subsection "\s-1ENTRY\s0"
T <cltEntry> is filled with the values for the new entry
.Sh "\s-1EXIT\s0"
.IX Subsection "\s-1EXIT\s0"
.Ip "\(bu" 4
.IX Item "\(bu"
T <cltEntry> is added to <*pclt>
.Ip "\(bu" 4
.IX Item "\(bu"
If an existing entry in <*pclt> matches <cltEntry> then
the entry's value is replaced.
.SH "NIFFIOCLTMakeDefaultEntries"
.IX Header "NIFFIOCLTMakeDefaultEntries"
Fill a chunk length table with \*(L"canned\*(R" entries.
.PP
.Vb 2
\&     RIFFIOSuccess
\&     NIFFIOCLTMakeDefaultEntries(NIFFIOChunkLengthTable *pclt)
.Ve
This routine is useful if you don't need a custom chunk length
table.  It fills an existing chunk length table with a complete set
of chunk length entries for each chunk in the NIFF specification.
.PP
TWB \- Should this include INFO chunks? probably.
.SH "NIFFIOCLTLookup"
.IX Header "NIFFIOCLTLookup"
Lookup a chunk length given an FOURCC.
.PP
.Vb 2
\&     RIFFIOSuccess
\&     NIFFIOCLTLookup(NIFFIOChunkLengthTable *pclt, niffChklentabEntry *cltEntryp)
.Ve
.Sh "\s-1ENTRY\s0"
.IX Subsection "\s-1ENTRY\s0"
T <cltEntryp->chunkName> contains the \s-1FOURCC\s0 to look up in <*pclt>.
.Sh "\s-1EXIT\s0"
.IX Subsection "\s-1EXIT\s0"
Fills in <cltEntryp->offsetOfFirstTag> according to <*pclt>.
.Sh "\s-1RETURN\s0"
.IX Subsection "\s-1RETURN\s0"
.Ip "*\s-1RIFFIO_OK\s0" 4
.IX Item "*\s-1RIFFIO_OK\s0"
if <cltEntryp->chunkName> was found.
.Ip "*\s-1RIFFIO_FAIL\s0" 4
.IX Item "*\s-1RIFFIO_FAIL\s0"
otherwise
.SH "NIFFIOCLTCount"
.IX Header "NIFFIOCLTCount"
Return the number of entries in a chunk length table
.PP
.Vb 2
\&     unsigned
\&     NIFFIOCLTCount(NIFFIOChunkLengthTable *pclt)
.Ve
.SH "NIFFIOCompareFOURCC"
.IX Header "NIFFIOCompareFOURCC"
Compare FOURCCs alphabetically.
.PP
.Vb 2
\&     int
\&     NIFFIOCompareFOURCC(FOURCC fccA, FOURCC fccB)
.Ve
.Sh "\s-1RETURN\s0"
.IX Subsection "\s-1RETURN\s0"
.Ip "\(bu" 4
.IX Item "\(bu"
<0 if a comes before b
.Ip "\(bu" 4
.IX Item "\(bu"
0 if a is the same as b
.Ip "\(bu" 4
.IX Item "\(bu"
>0 if a comes after b
.PP
The return results are compatible with \fIbsearch()\fR
and \fIstrcmp()\fR in the Standard C library.

.rn }` ''


niff-1.02/niffsdk1.02/niffio/src/lib/niffio/clt.c

#ifndef lint
static char rcsid[] = 
"$Id: clt.c,v 1.2 1996/06/15 02:53:52 tim Exp $";
#endif
/*
 * Public Domain 1995,1996 Timothy Butler
 *
 * THIS DOCUMENT IS PROVIDED "AS IS" AND WITHOUT ANY EXPRESS OR
 * IMPLIED WARRANTIES, INCLUDING, WITHOUT LIMITATION, THE IMPLIED
 * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
 *
 */
/***************************************************************************/
/*
 * NAME
 * ====
 * NIFFIO chunk length table routines.
 *
 * - NIFFIOCLTNew()
 * - NIFFIOCLTDelete()
 *
 * - NIFFIOCLTMakeEntry()
 * - NIFFIOCLTMakeDefaultEntries()
 * - NIFFIOCLTLookup()
 *
 * - NIFFIOCLTCount()
 * - NIFFIOCompareFOURCC()
 * 
 */
/***************************************************************************/

#include <assert.h>
#include <stdlib.h>
#include <string.h>

#include "niffio.h"

/*
 * Chunk length tables are implemented as RIFFIOFCCTables.
 * Store the FOURCC in the RIFFIOFCCTableEntry.fcc
 * Store the chunk length in the RIFFIOTableEntry.value.l
 */

/*
 * Provide a canned set of CLT entries in our own private array
 *
 * I count about 63 different chunk types 
 */
#define DEFAULT_CLT_LIM 62
static niffChklentabEntry _DefaultCLTEntries[]; 

static
RIFFIOSuccess
NIFFIOPCLTReallocate(NIFFIOChunkLengthTable *pclt, unsigned newReserved);


/***************************************************************************/
/*
 * NIFFIOCLTNew
 * ============
 * Return a new, empty NIFFIOChunkLengthTable, or null on failure.
 */
NIFFIOChunkLengthTable *
NIFFIOCLTNew()
/***************************************************************************/
{
    NIFFIOChunkLengthTable *pclt;  /* The table to return */
    
    pclt = RIFFIOFCCTableNew();
    
    return pclt;


}

/***************************************************************************/
/*
 * NIFFIOCLTDelete
 * ===============
 * Free the memory allocated to a chunk length table.
 */
void
NIFFIOCLTDelete(NIFFIOChunkLengthTable *pclt)
/***************************************************************************/
{
    assert(pclt != 0);

    RIFFIOFCCTableDelete(pclt);

}

/***************************************************************************/
/*
 * NIFFIOCLTMakeEntry
 * ==================
 * Make an entry into a chunk length table.
 */
RIFFIOSuccess
NIFFIOCLTMakeEntry(NIFFIOChunkLengthTable *pclt, 
                   niffChklentabEntry cltEntry)
/*
 * ENTRY
 * -----
 * T <cltEntry> is filled with the values for the new entry
 * 
 * EXIT
 * ----
 *  - T <cltEntry> is added to <*pclt>
 *  - If an existing entry in <*pclt> matches <cltEntry> then
 *    the entry's value is replaced.
 */
/***************************************************************************/
{
    char strModule[] = "NIFFIOCLTMakeEntry";
    char strChunkName[RIFFIO_FOURCC_LIM];
    
    RIFFIOFCCTableEntry  tableEntry; /* interface to RIFFIOFCCTableMakeEntry */
    
    RIFFIOFOURCCToString(cltEntry.chunkName, strChunkName);
    
    /*
     * Watch out for bogus chunkNames
     */
    if (!(RIFFIOFOURCCIsValid(cltEntry.chunkName)))
    {
        RIFFIOError(strModule, "Entry has an invalid chunkName <%s>",
                    strChunkName);
        return RIFFIO_FAIL;
    }
    
    tableEntry.fcc = cltEntry.chunkName;
    tableEntry.value.l = cltEntry.offsetOfFirstTag;
    
    return RIFFIOFCCTableMakeEntry(pclt, tableEntry);
    
}

/***************************************************************************/
/*
 * NIFFIOCLTMakeDefaultEntries
 * ===========================
 * Fill a chunk length table with "canned" entries.
 */
RIFFIOSuccess
NIFFIOCLTMakeDefaultEntries(NIFFIOChunkLengthTable *pclt)
/*
 * This routine is useful if you don't need a custom chunk length
 * table.  It fills an existing chunk length table with a complete set
 * of chunk length entries for each chunk in the NIFF specification.
 *
 * TWB - Should this include INFO chunks? probably.
 */
/**************************************************************************/
{

    niffChklentabEntry *pentry;

    assert(pclt);
    
    /* Insert our default entries until we 
     * reach the sentinel offset of -9999 
     */
    for (pentry = _DefaultCLTEntries; 
         pentry->offsetOfFirstTag != -9999; 
         pentry++)
    {
        if (! NIFFIOCLTMakeEntry(pclt, *pentry))
            return RIFFIO_FAIL;
    }
    
    return RIFFIO_OK;
    
}

/**************************************************************************/
/*
 * NIFFIOCLTLookup
 * ===============
 * Lookup a chunk length given an FOURCC.
 */
RIFFIOSuccess
NIFFIOCLTLookup(NIFFIOChunkLengthTable *pclt, niffChklentabEntry *cltEntryp)
/*
 * ENTRY
 * -----
 * T <cltEntryp->chunkName> contains the FOURCC to look up in <*pclt>.
 *
 * EXIT
 * ----
 * Fills in <cltEntryp->offsetOfFirstTag> according to <*pclt>.
 *
 * RETURN
 * ------
 *  RIFFIO_OK :
 *     if <cltEntryp->chunkName> was found.
 *
 *  RIFFIO_FAIL : 
 *     otherwise 
 */
/**************************************************************************/
{
    RIFFIOSuccess findResult;
    RIFFIOFCCTableEntry tableEntry;
    
    assert(pclt);
    assert(cltEntryp);
    
    tableEntry.fcc = cltEntryp->chunkName;
    findResult = RIFFIOFCCTableLookup(pclt, &tableEntry);
    
    if (findResult)
    {
        cltEntryp->offsetOfFirstTag = tableEntry.value.l;
    }
    
    return findResult;
}


/**************************************************************************/
/*
 * NIFFIOCLTCount
 * ==============
 * Return the number of entries in a chunk length table
 */
unsigned
NIFFIOCLTCount(NIFFIOChunkLengthTable *pclt)
/**************************************************************************/
{
    assert(pclt != 0);
    return RIFFIOFCCTableCount(pclt);
}

/**************************************************************************/
/* 
 * NIFFIOCompareFOURCC
 * ===================
 * Compare FOURCCs alphabetically.
 */
int
NIFFIOCompareFOURCC(FOURCC fccA, FOURCC fccB)
/*
 * RETURN
 * ------
 *       * <0 if a comes before b
 *       *  0 if a is the same as b
 *       * >0 if a comes after b
 *
 * The return results are compatible with bsearch()  
 * and strcmp() in the Standard C library.
 *
 */
/**************************************************************************/
{
    char strA[RIFFIO_FOURCC_LIM];
    char strB[RIFFIO_FOURCC_LIM];
    
    RIFFIOFOURCCToString(fccA, strA);
    RIFFIOFOURCCToString(fccB, strB);
    
    return strcmp(strA, strB);
    
}

#ifndef NDEBUG

/*
 * Make sure our canned CLT entries jive with a given chunk length table
 */
RIFFIOSuccess
NIFFIOCLTVerifyDefaults(NIFFIOChunkLengthTable *pclt)
{

    RIFFIOSuccess bFound;
    int i;     /* index into default chunk length table entries */
    niffChklentabEntry cltEntry; /* entry to lookup in clt */ 
    
    assert(pclt);
    
    for (i = 0; i < DEFAULT_CLT_LIM; i++)
    {
        char strFOURCC[RIFFIO_FOURCC_LIM];
        
        cltEntry.chunkName = _DefaultCLTEntries[i].chunkName;
        
        bFound = NIFFIOCLTLookup(pclt, &cltEntry);
        assert (bFound);
        assert (cltEntry.chunkName == _DefaultCLTEntries[i].chunkName);
        assert (cltEntry.offsetOfFirstTag 
                == _DefaultCLTEntries[i].offsetOfFirstTag);
        RIFFIOFOURCCToString(cltEntry.chunkName, strFOURCC);
        printf("<%s> %ld\n", strFOURCC, cltEntry.offsetOfFirstTag);
        
    }
    
    return RIFFIO_OK;
}


#endif

/*
 * Here are our canned CLT entries
 * 
 * WARNING
 * -------
 * If you add an entry, you must change DEFAULT_CLT_LIM.
 * A better way to do this would be to store a sentinel value
 * at the end of the array.
 */
#define DEFCLTENTRY(name) \
{niffckid##name, niffcklen##name},

static 
niffChklentabEntry _DefaultCLTEntries[] = 
{
    DEFCLTENTRY(ChnkLenTable)
    DEFCLTENTRY(DefaultValues)
    DEFCLTENTRY(EpsGraphic)
    DEFCLTENTRY(FontDescription)
    DEFCLTENTRY(NiffInfo)
    DEFCLTENTRY(Part)
    DEFCLTENTRY(PsType1Font)
    DEFCLTENTRY(PsType3Font)
    DEFCLTENTRY(StaffGrouping)
    DEFCLTENTRY(StringTable)
    
    DEFCLTENTRY(PageHeader)
    DEFCLTENTRY(StaffHeader)
    DEFCLTENTRY(SystemHeader)
    DEFCLTENTRY(TimeSlice)
    DEFCLTENTRY(Accidental)
    DEFCLTENTRY(AltEndingGraphic)
    DEFCLTENTRY(Arpeggio)
    DEFCLTENTRY(Articulation)
    DEFCLTENTRY(AugDot)
    DEFCLTENTRY(Barline)
    
    DEFCLTENTRY(Beam)
    DEFCLTENTRY(ChordSymbol)
    DEFCLTENTRY(Clef)
    DEFCLTENTRY(CustomGraphicChk)
    DEFCLTENTRY(Dynamic)
    DEFCLTENTRY(FiguredBass)
    DEFCLTENTRY(Fingering)
    DEFCLTENTRY(Glissando)
    DEFCLTENTRY(GuitarGrid)
    DEFCLTENTRY(GuitarTabNum)
    
    DEFCLTENTRY(Hairpin)
    DEFCLTENTRY(HarpPedal)
    DEFCLTENTRY(KeySignature)
    DEFCLTENTRY(KeySignNonstandard)
    DEFCLTENTRY(Line)
    DEFCLTENTRY(Lyric)
    DEFCLTENTRY(MeasureNumbering)
    DEFCLTENTRY(MidiDataStream)
    DEFCLTENTRY(FontSymbol)
    DEFCLTENTRY(Notehead)
    
    DEFCLTENTRY(OctaveSign)
    DEFCLTENTRY(Ornament)
    DEFCLTENTRY(Parenthesis)
    DEFCLTENTRY(PedalPiano)
    DEFCLTENTRY(PedalOrgan)
    DEFCLTENTRY(Portamento)
    DEFCLTENTRY(RehearsalMark)
    DEFCLTENTRY(RepeatSign)
    DEFCLTENTRY(Rest)
    DEFCLTENTRY(Slur)
    
    DEFCLTENTRY(Stem)
    DEFCLTENTRY(SystemSeparation)
    DEFCLTENTRY(TagActivate)
    DEFCLTENTRY(TagInactivate)
    DEFCLTENTRY(TempoMarking)
    DEFCLTENTRY(TempoMarkNonstandard)
    DEFCLTENTRY(Text)
    DEFCLTENTRY(Tie)
    DEFCLTENTRY(TimeSignature)
    DEFCLTENTRY(TimeSigNonstandard)
    
    DEFCLTENTRY(Tremolo)
    DEFCLTENTRY(Tuplet)

    { RIFFIO_FOURCC_IARL , -1 },
    { RIFFIO_FOURCC_IART , -1 },
    { RIFFIO_FOURCC_ICMS , -1 },
    { RIFFIO_FOURCC_ICMT , -1 },
    { RIFFIO_FOURCC_ICOP , -1 },
    { RIFFIO_FOURCC_ICRD , -1 },
    { RIFFIO_FOURCC_ICRP , -1 },
    { RIFFIO_FOURCC_IDIM , -1 },
    { RIFFIO_FOURCC_IDPI , -1 },
    { RIFFIO_FOURCC_IENG , -1 },
    { RIFFIO_FOURCC_IGNR , -1 },
    { RIFFIO_FOURCC_IKEY , -1 },
    { RIFFIO_FOURCC_ILGT , -1 },
    { RIFFIO_FOURCC_IMED , -1 },
    { RIFFIO_FOURCC_INAM , -1 },
    { RIFFIO_FOURCC_IPLT , -1 },
    { RIFFIO_FOURCC_IPRD , -1 },
    { RIFFIO_FOURCC_ISBJ , -1 },
    { RIFFIO_FOURCC_ISFT , -1 },
    { RIFFIO_FOURCC_ISHP , -1 },
    { RIFFIO_FOURCC_ISRC , -1 },
    { RIFFIO_FOURCC_ISRF , -1 },

    { RIFFIOMAKEFOURCC(' ', ' ', ' ', ' '), -9999 } /* Sentinel value,
                                                     *  don't process this */
};








